sdk/communication/AzureCommunicationChat/Source/Registrar/RegistrarClient.swift

```diff
@@ -80,79 +99,113 @@ internal class RegistrarClient {
         return try? JSONEncoder().encode(data)
     }
 
-    /// Sets the authentication header on a Registrar request.
+    /// Create a Registrar POST request.
     /// - Parameters:
-    ///   - request: The  HTTP request.
-    ///   - completionHandler: Returns the request, or an error if the request failed to be authenticated.
-    private func setAuthHeader(
-        on request: URLRequest,
-        completionHandler: @escaping (URLRequest?, AzureError?) -> Void
+    ///   - clientDescription: RegistrarClientDescription.
+    ///   - transports: RegistrarTranports, a transport contains the APNS token as the path.
+    ///   - completionHandler: Returns the POST request.
+    private func createPostRequest(
+        with clientDescription: RegistrarClientDescription,
+        for transports: [RegistrarTransport],
+        completionHandler: @escaping (HTTPRequest?, AzureError?) -> Void
     ) {
-        credential.token { token, error in
-            guard let skypeToken = token?.token else {
-                completionHandler(
-                    nil,
-                    AzureError.client("Failed to get token from CommunicationTokenCredential.", error)
-                )
+        guard let data = createPostData(with: clientDescription, for: transports) else {
+            completionHandler(nil, AzureError.client("Failed to serialize POST request body."))
+            return
+        }
```

Comment:
I don't see what relevance the transport stage would have in serializing POST bodies.

Reply:
This is similar to the above reply. This transport stage is not (B) the one that is required to initialize the PipelineClient, but (A) the one contains the APNS token required in Push Notification Feature. So we pass it here to create the PostData. If the data is not created successfully, there can be an issue in this line - "return try? JSONEncoder().encode(data)" inside the function createPostData(). So we throw the error as "Failed to serialize POST request body".